tools/preprocess_dataset_mpi.py

```diff
@@ -391,6 +346,11 @@ def get_start_end(num, rank, num_ranks):
 def get_filename(args, key, rank=None):
     pathname = args.output_prefix
 
+    # redirect per-rank file to scratch dir if defined
+    if args.scratch and rank is not None:
```

Comment:
```suggestion
    if args.scratch is not None and rank is not None:
```

Sorry just a personal preference, but "" evaluates to False too, though has a different meaning than None.

Reply:
Fixed this one.